scripts/addons.py

```diff
@@ -0,0 +1,267 @@
+                    self._parse_organization_repo(_args[0])
+                    self._set_branch(_args[1])
+        elif _len_args == 3:
+            if self._check_is_url(_args[1]):
+                # repo OR organization/repo AND url AND branch
+                self._parse_organization_repo(_args[0])
+                self._parse_url(_args[1])
+                self._set_branch(_args[2])
+                self.folder_name = _args[0]
+
+    @property
+    def path(self):
+        return '%s%s' % (EXTRA_ADDONS_PATH, self.folder_name)
+
+    @property
+    def resolve_url(self):
+        _resolve_url = '%s%s/%s/%s.git' % (
+            self.scheme,
+            self.git_repo_host,
+            self.organization,
+            self.repository
+        )
+        return _resolve_url
+
+    @property
+    def download_cmd(self):
+        if self.branch:
+            cmd = 'git clone -b %s %s %s' % (
+                self.branch, self.resolve_url, self.path
+            )
+        else:
+            cmd = 'git clone %s %s' % (
+                self.resolve_url, self.path
+            )
+        return cmd.split()
+
+    @property
+    def update_cmd(self):
+        if self.branch:
+            cmd = 'git --git-dir=%s/.git --work-tree=%s pull origin %s' % (
+                self.path, self.path, self.branch
+            )
+            return cmd.split()
+        else:
+            self.get_branch_name()
+            cmd = 'git --git-dir=%s/.git --work-tree=%s pull origin %s' % (
+                self.path, self.path, self.branch
+            )
+            return cmd.split()
+
+    def download(self):
+        if self.path in ADDONS_PATH:
+            return
+        if os.path.exists(self.path):
+            fetch_oca_dependencies = os.getenv('FETCH_OCA_DEPENDENCIES')
+            if fetch_oca_dependencies == 'FALSE':
+                self.get_branch_name()
+            else:
+                print('PULL: %s %s' % (self.path, self.branch))
+                cmd = self.update_cmd
+                call(cmd)
+        else:
+            print('CLONE: %s %s' % (self.path, self.branch))
+            call(self.download_cmd)
+            self.get_branch_name()
+
+        ADDONS_PATH.append(self.path)
+        self.download_dependency()
+
+    def download_dependency(self):
+        filename = '%s/%s' % (self.path, DEPENDENCIES_FILE)
+        if not os.path.exists(filename):
+            return
+        repo_list = []
+        with open(filename) as f:
+            for line in f:
+                l = line.strip('\n').strip()
+                if l.startswith('#') or not l:
+                    continue
+                repo_list.append(Repo(l, self))
+        f.close()
+        for repo in repo_list:
+            repo.download()
+
+
+def write_addons_path():
+    contains_addons_path = False
+    with open(ODOO_CONF, 'r') as conf_file:
+        for line in conf_file:
+            if re.match(r'^addons_path*=*', line):
```

Comment:
Are you sure it won't match `addons_path_whatever =`? IMO it should be `\s*` rather than just `*` (but it doesn't look like POSIX regex so I'm not sure)